src/swarm-tui/Swarm/TUI/Model/StateUpdate.hs

```diff
@@ -91,6 +94,35 @@ import Swarm.TUI.View.Structure qualified as SR
 import Swarm.Util (listEnums)
 import Swarm.Util.Effect (asExceptT, withThrow)
 import System.Clock
+import System.Exit (exitFailure)
+
+createEventHandlers :: KeyConfig SwarmEvent -> IO EventHandlers
+createEventHandlers config = do
+  mainHandler <- buildDispatcher mainEventHandlers
+  replHandler <- buildDispatcher replEventHandlers
+  return EventHandlers {..}
+ where
+  -- this error handling code is taken from the brick demo app:
+  -- https://github.com/jtdaugherty/brick/blob/764e66897/programs/CustomKeybindingDemo.hs#L216
+  buildDispatcher handlers = case keyDispatcher config handlers of
+    Right d -> return d
+    Left collisions -> do
+      putStrLn "Error: some key events have the same keys bound to them."
+      forM_ collisions $ \(b, hs) -> do
+        T.putStrLn $ "Handlers with the '" <> BK.ppBinding b <> "' binding:"
+        forM_ hs $ \h -> do
+          let trigger = case BK.kehEventTrigger $ BK.khHandler h of
+                ByKey k -> "triggered by the key '" <> BK.ppBinding k <> "'"
+                ByEvent e -> "triggered by the event '" <> fromMaybe "<unknown>" (BK.keyEventName swarmEvents e) <> "'"
+              desc = BK.handlerDescription $ BK.kehHandler $ BK.khHandler h
+          T.putStrLn $ "  " <> desc <> " (" <> trigger <> ")"
+      exitFailure
+
+initKeyHandlingState :: IO KeyEventHandlingState
```

Comment:
Could be nice to give these types like `Has (Lift IO) sig m => m KeyEventHandlingState`.  Then you can call them without `sendIO`, and they become a bit more flexible (e.g. maybe later we realize they need some other effect).

Reply:
Done, I will use it to plug into existing SystemFailure handling. 👍 

I am not sure whether I should use Throw or Accum, maybe it's better to only have the failures as warnings...

Reply:
I think `Throw` is fine, if the user's keybinding config cannot be parsed I think they would want to fix that before running the game.  Especially since right now warnings are a bit hidden (they go into `Messages` but I never think to look there).